spacy/gold.pyx

```diff
@@ -355,25 +402,37 @@ class GoldCorpus(object):
         else:
             docs = []
             raw_text, paragraph_tuples = make_orth_variants(nlp, None, paragraph_tuples, orth_variant_level=orth_variant_level)
-            return [Doc(nlp.vocab, words=add_noise(sent_tuples[1], noise_level))
-                    for (sent_tuples, brackets) in paragraph_tuples], paragraph_tuples
-
+            dd = [Doc(nlp.vocab, words=add_noise(sent_tuples[1], noise_level)) for (sent_tuples, brackets) in paragraph_tuples]
+            return dd, paragraph_tuples
 
     @classmethod
-    def _make_golds(cls, docs, paragraph_tuples, make_projective, ignore_misaligned=False):
+    def _make_golds(cls, docs, paragraph_tuples, make_projective,
+                    ignore_misaligned=False,
+                    ignore_errors=False,
+                    doc_id=None,
+                    errors=None):
         if len(docs) != len(paragraph_tuples):
             n_annots = len(paragraph_tuples)
             raise ValueError(Errors.E070.format(n_docs=len(docs), n_annots=n_annots))
         golds = []
+        error_occurred = False
         for doc, (sent_tuples, (cats, brackets)) in zip(docs, paragraph_tuples):
             try:
-                gold = GoldParse.from_annot_tuples(doc, sent_tuples, cats=cats,
-                    make_projective=make_projective)
+                gold = GoldParse.from_annot_tuples(doc, sent_tuples, cats=cats, make_projective=make_projective)
             except AlignmentError:
                 if ignore_misaligned:
                     gold = None
                 else:
                     raise
+            except ValueError as e:
+                if errors is not None and not error_occurred:
+                    t = (doc_id, e)
+                    errors.append(t)
+                    error_occurred = True
```

Comment:
is the purpose of `error_occurred` to only record one error per doc? If so, shouldn't it be set to `False` as the first line within the `for` loop rather than before it ?